data/transactions/transaction.go

```diff
@@ -177,31 +178,81 @@ func (tx Transaction) ToBeHashed() (protocol.HashID, []byte) {
 	return protocol.Transaction, protocol.Encode(&tx)
 }
 
+// txAllocSize returns the max possible size of a transaction without state proof fields.
+// It is used to preallocate a buffer for encoding a transaction.
+func txAllocSize() int {
```

Comment:
We can make this a constant and have the function be a test ensuring the size of the constant is correct but I guess in ideal world we should only call it a fixed number of times since once the pool is populated we should only be-reusing already initialized backing slices.

Reply:
Yes, expectation is to have it called not too often.